### ISSUE TITLE
Syntax error for TypeScript generics

### ISSUE THREAD

**Issue by bajtos** [alecthomas/chroma#425]
**Describe the bug**

Chroma seems to not understand TypeScript generics with multiple template arguments and/or template arguments using a generic type too.

Consider the following TypeScript code:

```ts
export class AuditingRepository<
  T extends Entity,
  ID,
  Relations extends object = {}
> extends DefaultCrudRepository<T, ID, Relations> {
}
```

Chroma v0.8.2 and Hugo v0.78.2 render the code with a comma (`,`) characters highlighted in red - perhaps suggesting an error?

![chroma-typescript-bug](https://user-images.githubusercontent.com/1140553/99884655-32f68080-2c30-11eb-80a8-6b8ad2618af2.png)

**To Reproduce**

Save the following text to `bug.ts`.

```ts
export class AuditingRepository<
  T extends Entity,
  ID,
  Relations extends object = {}
> extends DefaultCrudRepository<T, ID, Relations> {
}
```

Run chroma as follows:

```
chroma -s autumn --html --html-lines --html-lines-table --html-inline-styles bug.ts > bug.html
```

Open the produced HTML file in a Chromium-based browers. You should see the same output as on my screenshot posted above.

**Comment by fredrare** [comment]
### Cause
This happens because the TypeScript lexer also detects jsx. In the case of generics, they are not actually being treated as such, but rather as JSX elements. For this reason, the `,` is not allowed. Note that this is also an issue in case you want to add something like `Required<Pick<T, Key>>`, because the `<` character is not allowed inside a JSX tag for obvious reasons. This may happen with any TypeScript specific syntax that would not be correct in jsx.

<img width="359" alt="Screenshot 2024-09-20 at 9 28 05 PM" src="https://github.com/user-attachments/assets/b03c2c8a-3278-438b-8803-00eb62cc4a17">

### Solution
Without needing to modify the lexer, you can bypass this error simply adding a space between `<` and the generic you want to use. The JSX rules only apply when there are no spaces between that sign and the tag name, so it treats them as 2 separate entities: `Operator` and `NameOther`. Of course, given that this solution leaves you with ✨*asymmetrical*✨ generics, also add an space before `>` to please your eyes.

<img width="392" alt="Screenshot 2024-09-20 at 9 30 13 PM" src="https://github.com/user-attachments/assets/89289476-c92a-41df-b893-5ab289fa96aa">

**Comment by alecthomas** [comment]
Fixed in #1002
